### ISSUE TITLE
Any way to increase iteration speed?

### ISSUE THREAD

**Issue by yay** [iced-rs/iced#638]
Hello, I'm playing with the basic examples, like a single button changing the window title on click. But every time I run the app after making a change to the title string it takes a little over 10 seconds to recompile (on a 16" i9 MBP).

This is really disappointing coming from both web apps and native apps (Swift/SwiftUI, Delphi/VCL or C++/Qt based). At least when we are talking about basic apps like this.

Any way to improve this? I know the native frameworks I mentioned use dynamic libraries for the framework code while Rust has to recompile from source, but I'm really wondering about solutions rather than reasons for slow compile times.

I'm using the recently released iced 0.2 and Rust 1.48 on macOS 11.1

**Comment by yaskhan** [comment]
Maybe ramdisk will help you?

**Comment by manokara** [comment]
It's true that Rust has a slow compile time for several reasons, but 10 seconds for that is way too much. I'm suspicious it could be a hardware issue ([from what I gathered](https://appleinsider.com/articles/18/07/18/tested-thermal-conditions-in-the-2018-i9-macbook-pro-dramatically-hampering-performance), not sure if the 16" model has the same issue) as my air-cooled Ryzen 5 2400G recompiles my quite substantial [hexview demo](https://github.com/manokara/iced_widgets/tree/main/examples/hexview/) in half that time when changing a couple lines, sometimes even a little bit less.

But there's a way to reduce compile times a little bit, described [here](https://bevyengine.org/learn/book/getting-started/setup/#enable-fast-compiles-optional). Might be worth a try.

**Comment by yay** [comment]
I don't believe it's thermals related, the 16" model has a redesigned body with much larger air intakes and bigger fans to improve airflow. But it shouldn't make a difference either way compiling what is essentially a Hello World app from idle.

I don't believe it's IO bound either, I had a Ryzen 3900X PC with 1TB Evo 970 NVMe SSD that compiled our whole monorepo at the same speed (slightly slower actually, maybe because Windows...) as this laptop, so it's not IO bound either.

**Comment by alex13sh** [comment]
Use cargo build incremental 
https://doc.rust-lang.org/cargo/reference/config.html

**Comment by yay** [comment]
Adding
```
[build]
incremental = true
```
to `./cargo/config.toml` doesn't seem to do anything:
<img width="476" alt="Screen Shot 2020-12-02 at 19 41 34" src="https://user-images.githubusercontent.com/1799756/100923024-82616a00-34d6-11eb-9fd5-811c8043d7d3.png">

**Comment by manokara** [comment]
That's because in debug builds incremental is enabled by default. Have you tried the linker approach I mentioned in my previous post? Still, I can't see a reason why an i9 would be slower than a 2nd gen Ryzen, even if it's a mobile chip. What could possibly be going on?

**Comment by yay** [comment]
Thanks @manokara, I just tried that (was a bit busy at the time). With rustc 1.50 nightly and zld 1.2.1 the debug build now takes 6.29s, which is obviously a massive difference 👍, although I wouldn't mind improving it further :)

**Comment by yay** [comment]
One thing I noticed is that after upgrading to macOS 11.0.1 VS Code got slower because of this issue: https://github.com/microsoft/vscode/issues/105446#issuecomment-729274600
I wonder if it plays a part here.

**Comment by cryptoquick** [comment]
These tweaks worked really well for me, check out the PR I referenced :point_up:

**Comment by yay** [comment]
Thanks @cryptoquick, they worked for me too, that's what I was referring to here: https://github.com/hecrj/iced/issues/638#issuecomment-737570512

**Comment by cryptoquick** [comment]
@yay If you're currently using a Mac that costs less than $5000, you're almost definitely not using a comparable CPU. My CPU has a TDP of 105W. However, I also noticed that your build key recommendation seemed to be ignored in my screenshot, as I understand @manokara pointed out. Are you using zld, like the bevy instructions indicated? You also have to add the .cargo/config.toml, and be using nightly.

**Comment by yay** [comment]
@cryptoquick Haha, no worries, sadly, I am using a MBP that is worth £4000 with i9-9980H inside. And reading tweets like this makes me think I'll have to upgrade next year: https://twitter.com/rikarends/status/1328598935380910082

**Comment by cryptoquick** [comment]
@yay Hold out for the M1X :wink:

**Comment by yay** [comment]
@cryptoquick I did :) and it's a big help! Now waiting for [mold](https://github.com/rui314/mold) to come to macOS 😛

**Comment by 13r0ck** [comment]
https://github.com/iced-rs/iced/issues/21
